### PR TITLE
Ensure sql.go and related are always regenerated with 'go generate'.

### PR DESCRIPTION
**Pull request by knz**
As found via #4537 and discussed on eng@.

[Review on Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/4590)

### CODE REVIEW

File: sql/parser/Makefile [hunk 1]
```diff
@@ -1,5 +1,6 @@
-.PHONY: all
-all: sql.go keywords.go reserved_keywords.go
+TARGETS = sql.go keywords.go reserved_keywords.go
+.PHONY: all $(TARGETS)
```
Comment (petermattis) [inline sql/parser/Makefile]: The targets (`{sql,keywords,reserved_keywords}.go`) are not phony. They are real files for which we want make to compare the timestamps. I missed the discussion of why this is necessary. Care to elaborate? If I stumbled across a Makefile containing the above I would fix it, so a comment is definitely necessary to explain this peculiar usage of `.PHONY`.

Comment (knz) [inline sql/parser/Makefile]: @petermattis The problem is that if you have a commit which has generated sql.go from sql.y incorrectly (e.g. wrong version of Go), "make" will not see a need to re-generate.

Given we only run "make" in sql/parser, or "go generate", by hand it seems reasonable to force re-generation at all times.

Comment (petermattis) [inline sql/parser/Makefile]: Yeah, I suppose it makes sense to always regenerate `sql.go` when make is invoked. We've taken the style of placing `.PHONY` declarations immediately in front of the targets they apply to (see the top-level `Makefile` for examples). So:

```
.PHONY: sql.go
sql.go:
   ...

.PHONY: keywords.go
keywords.go:
   ...
```

Also, remove the dependency of `sql.go` on `sql.y`. Now that `sql.go` is a phony target that isn't necessary and slightly confusing. Ditto for the dependencies for `keywords.go` and `reserved_keywords.go`.

Note we could have also fixed this by making `sql.go` depend on `sql_union.sh` and `$(which yacc)`. Seems safer to always rebuild, though.

**Comment by knz** [comment]
I have spread out `.PHONY` as suggested. Note however that I kept the dependencies listed, because if they are broken (e.g. file erased by mistake) the error message by make is more pleasant to read than the one by the shell due to the invalid redirect.

**Comment by petermattis** [comment]
LGTM

Not sure if we should be optimizing error messages for very unusual circumstances such as deleting `sql.y`.

**Pull request merged**